share/completions/eopkg.fish

```diff
@@ -0,0 +1,153 @@
+        end
+    end
+    return 1
+end
+
+# Eopkg subcommand
+function __fish_eopkg_subcommand
+    set subcommand $argv[1]
+    set -e argv[1]
+    complete -f -c eopkg -n '__fish_use_subcommand' -a $subcommand $argv
+end
+
+function __fish_eopkg_subcommand_with_shortcut
+    set subcommand $argv[1]
+    set -e argv[1]
+    set shortcut $argv[1]
+    set -e argv[1]
+    complete -f -c eopkg -n '__fish_use_subcommand' -a $subcommand $argv
+    complete -f -c eopkg -n '__fish_use_subcommand' -a $shortcut $argv
+end
+
+function __fish_eopkg_using_subcommand -d 'Test if given subcommand is used'
+    for i in (commandline -poc)
+        if contains -- $i $argv
+            return 0
+        end
+    end
+    return 1
+end
+
+# Eopkg subcommand's option
+function __fish_eopkg_option
+    set subcommand $argv[1]
+    set -e argv[1]
+    complete -f -c eopkg -n "__fish_eopkg_using_subcommand $subcommand" $argv
+end
+
+function __fish_eopkg_option_with_shortcut
+    set subcommand $argv[1]
+    set -e argv[1]
+    set shortcut $argv[1]
+    set -e argv[1]
+    complete -f -c eopkg -n "__fish_eopkg_using_subcommand $subcommand" $argv
+    complete -f -c eopkg -n "__fish_eopkg_using_subcommand $shortcut" $argv
+end
+
+# Print additional completion
+function __fish_eopkg_print_components -d "Print list of components"
+    eopkg list-components -N | cut -d ' ' -f 1
+end
+
+function __fish_eopkg_print_repos -d "Print list of repositories"
+    eopkg list-repo -N | grep active | cut -d ' ' -f 1
+end
+
+# Setup additional completion
+complete -f -c eopkg -n '__fish_eopkg_package_ok; and not __fish_eopkg_component_ok' -a "(__fish_print_packages)" -d "package"
+complete -f -c eopkg -n '__fish_eopkg_component_ok' -a "(__fish_eopkg_print_components)" -d "component"
+complete -f -c eopkg -n '__fish_eopkg_repo_ok' -a "(__fish_eopkg_print_repos)" -d "repository"
+
+# Setup eopkg subcommand with shortcut
+__fish_eopkg_subcommand_with_shortcut upgrade up -d "Upgrades eopkg packages"
+__fish_eopkg_option_with_shortcut upgrade up -l security-only -d "Security related package upgrades only"
+__fish_eopkg_option_with_shortcut upgrade up -s n -l dry-run -d "Do not perform any action, just show what would be done"
+__fish_eopkg_option_with_shortcut upgrade up -s c -l component -f -d "Upgrade component's and recursive components' packages"
+
+__fish_eopkg_subcommand_with_shortcut install it -d "Install eopkg packages"
+__fish_eopkg_option_with_shortcut install it -l reinstall -d "Reinstall already installed packages"
+__fish_eopkg_option_with_shortcut install it -s n -l dry-run -d "Do not perform any action, just show what would be done"
+__fish_eopkg_option_with_shortcut install it -s c -l component -f -d "Install component's and recursive components' packages"
+
+__fish_eopkg_subcommand_with_shortcut remove rm -d "Remove eopkg packages"
+__fish_eopkg_option_with_shortcut remove rm -l purge -d "Removes everything including changed config files of the package"
```

Comment:
You're using the imperative mood elsewhere - "Remove ...". Do that here as well:

"Remove everything including changed config files"? Or "Remove changed config files as well"?